### ISSUE TITLE
Support `COPY --link --from` instructions

### ISSUE THREAD

**Issue by kevcube** [hadolint/hadolint#826]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

no warnings or errors, this is valid syntax as of dockerfile version 1.4.0

### Actual behavior

error from hadolint

### Steps to reproduce the behavior

see dockerfile

**Output of `hadolint --version`**

```bash
2.9.3
```

**Dockerfile** (if relevant)

```Dockerfile
# syntax=docker/dockerfile:1.4
FROM node:16-alpine AS deps
WORKDIR /app
COPY --link package.json yarn.lock .npmrc ./
RUN --mount=type=cache,id=yarn,target=/usr/local/share/.cache/yarn/v6 \
    --mount=type=secret,id=npmrc,target=/root/.npmrc,required=true \
    yarn install --frozen-lockfile --non-interactive --cache-folder=/usr/local/share/.cache/yarn/v6

# Rebuild the source code only when needed
FROM node:16-alpine AS builder
WORKDIR /app
COPY --link --from=deps /app/node_modules ./node_modules
```

**Comment by wholtz** [comment]
The `--from` option is not required to get the false positive error message:

```Dockerfile
## syntax=docker/dockerfile:1.4
FROM scratch
COPY --link ./Dockerfile /Dockerfile
```

```bash
$ docker run --rm -i hadolint/hadolint < Dockerfile
-:3 DL3021 error: COPY with more than 2 arguments requires the last argument to end with /
$ docker run --rm -i hadolint/hadolint hadolint --version
Haskell Dockerfile Linter 2.10.0
```

**Comment by modem7** [comment]
Same issue with `ADD` as well.

**Comment by miigotu** [comment]
Also hits on `RUN --mount ...`
```
RUN --mount=type=tmpfs,target="$CARGO_HOME" pip install --upgrade "sickchill[speedups]";
```

**Comment by maxhelias** [comment]
This seems to be fixed since version 2.11 which includes https://github.com/hadolint/language-docker/pull/84
